src/lib/operators/jit_operator/jit_operations.hpp

```diff
@@ -0,0 +1,195 @@
+#pragma once
+
+#include <boost/preprocessor/seq/for_each_product.hpp>
+#include <boost/preprocessor/tuple/elem.hpp>
+
+#include <cmath>
+
+#include "jit_types.hpp"
+#include "operators/table_scan/like_table_scan_impl.hpp"
+
+namespace opossum {
+
+/* This file contains the type dispatching mechanisms that allow generic operations on JitMaterializedValues.
+ *
+ * Each binary operation requires three JitMaterializedValues - a left input (lhs), a right input (rhs) and an
+ * output (result). Each value has one of 6 data types (bool, int, long, float, double, std::string) and can be
```

Comment:
We try to keep the list of supported data types central, so I wouldn't explicitly list them here.